sympy/core/compatibility.py

```diff
@@ -857,6 +857,6 @@ def cache_clear():
     return decorating_function
 ### End of backported lru_cache
 
-if sys.version_info[:2] >= (3, 3):
+if sys.version_info[:2] >= (3, 6):
```

Comment:
This is not the place for the matchpy version test. It could be in `rubi/__init__.py` to control importing. (The import of lru_cache should not be changed.)